### PR TITLE
DOC add section on interpreting error/warnings

### PR DESCRIPTION
**Pull request by lucyleeow**
Related to #725 

Also amended some doc nitpicks.

### CODE REVIEW

**Comment by lucyleeow** [comment]
@larsoner last thing before I do the release!

File: doc/advanced.rst [hunk 1]
```diff
@@ -89,6 +89,38 @@ The above directive inserts the following block:
...
+error::
+
+    File "<full_path>/examples/no_output/plot_raise.py", line 27, in <module>
+        iae
+    NameError: name 'iae' is not defined
+
+Problems in the text (rST) blocks of the gallery Python files will result
+in warnings or errors when Sphinx is converting the generated ``.rst`` files
+to HTML. These will also be printed in pink, after code block errors, during
+building of the documentation. In this case, the ``.rst`` file path and
```
Comment (larsoner) [inline doc/advanced.rst:111]: I would say "by Sphinx" or something to make it clear it's the upstream project emitting those

File: doc/advanced.rst [hunk 2]
```diff
@@ -89,6 +89,38 @@ The above directive inserts the following block:
...
+to amend the original ``.py`` file, **not** the generated ``.rst`` file.
+To figure out where the problem is, you will need to match the content of the
+``.rst`` file at the line number printed to the original ``.py`` file.
+
+Example ``.rst`` warning::
+
+    <full_path>/auto_examples/plot_example.rst:19: WARNING: Explicit markup ends without a blank line; unexpected unindent.
+
+The warning above was occurred in line 19 in ``plot_example.rst``. The
+original ``plot_example.py`` file will need to be amended to fix.
```
Comment (larsoner) [inline doc/advanced.rst:122]: Then mention that SG only (re)builds new, changed, or failed examples, so doing another `sphinx-build` should rebuild just the modified example, allowing for quick iteration?

**Comment by larsoner** [comment]
Thanks @lucyleeow

**Pull request merged**